clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -3887,6 +3887,18 @@ void handleFailure(Throwable throwable) {
         return calls;
     }
 
+    private List<Call> getListOffsetsCalls(MetadataOperationContext<ListOffsetsResultInfo, ListOffsetsOptions> context,
+                                           Map<TopicPartition, OffsetSpec> topicPartitionOffsets,
+                                           Map<TopicPartition, KafkaFutureImpl<ListOffsetsResultInfo>> futures,
+                                           Call failedCall) {
```

Comment:
If I'm understanding this code correctly, the `failedCall` will update the `tries` and `nextAllowedTryMs` for all calls, but we only want to update it for the call that specifically failed, not for all calls. This patch would not work correctly for a scenario where there are multiple topic partitions, each with different leaders, and one of the topic partitions failed because the leader for that topic partition was not available. I think this patch needs to be reworked in order to account for the fact that not all calls in the list would fail (as each call is made based on the number of leaders in the cluster)

Reply:
As we discussed, this code piece probably won't increase the backoff and numtries for everycall. I think there’s one call per leader. In the call for every leader, only the failed topic partitions’ offsets will be requested.